src/components/application_manager/rpc_plugins/vehicle_info_plugin/test/commands/mobile/subscribe_vehicle_request_test.cc

```diff
@@ -0,0 +1,325 @@
+#include "vehicle_info_plugin/commands/vi_command_request_test.h"
+#include "vehicle_info_plugin/vehicle_info_app_extension.h"
+#include "vehicle_info_plugin/vehicle_info_plugin.h"
+
+#include "application_manager/mock_rpc_handler.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace mobile_commands_test {
+namespace subscribe_vehicle_data_request {
+
+namespace am = ::application_manager;
+namespace mobile_result = mobile_apis::Result;
+
+using ::testing::_;
+
+using am::commands::MessageSharedPtr;
+using vehicle_info_plugin::commands::SubscribeVehicleDataRequest;
+
+typedef std::shared_ptr<SubscribeVehicleDataRequest> CommandPtr;
+
+namespace {
+const uint32_t kConnectionKey = 1u;
+const std::string kMsgParamKey = "test_key";
+const mobile_apis::VehicleDataType::eType kVehicleType =
+    mobile_apis::VehicleDataType::VEHICLEDATA_WINDOWSTATUS;
+const std::string kVehicleTypeStr = am::strings::window_status;
+}  // namespace
+
+class SubscribeVehicleRequestTest
+    : public VICommandRequestTest<CommandsTestMocks::kIsNice> {
+ public:
+  SubscribeVehicleRequestTest()
+      : mock_app_(CreateMockApp())
+      , vi_app_extension_ptr_(
+            std::make_shared<vehicle_info_plugin::VehicleInfoAppExtension>(
+                vi_plugin_, *mock_app_))
+      , app_set_lock_ptr_(std::make_shared<sync_primitives::Lock>())
+      , mock_last_state_(std::make_shared<resumption::LastStateWrapperImpl>(
+            std::make_shared<resumption_test::MockLastState>())) {}
+
+ protected:
+  void SubscribeSuccessfully();
+  void SetUp() OVERRIDE {
+    ON_CALL(app_mngr_, GetRPCHandler())
+        .WillByDefault(ReturnRef(mock_rpc_handler_));
+
+    vi_plugin_.Init(app_mngr_,
+                    mock_rpc_service_,
+                    mock_hmi_capabilities_,
+                    mock_policy_handler_,
+                    mock_last_state_);
+    ON_CALL(*mock_app_, AddExtension(vi_app_extension_ptr_));
+    vi_plugin_.OnApplicationEvent(application_manager::plugin_manager::
+                                      ApplicationEvent::kApplicationRegistered,
+                                  mock_app_);
+    ON_CALL(*mock_app_,
+            QueryInterface(vehicle_info_plugin::VehicleInfoAppExtension::
+                               VehicleInfoAppExtensionUID))
+        .WillByDefault(Return(vi_app_extension_ptr_));
+    ON_CALL(*mock_app_, app_id()).WillByDefault(Return(kConnectionKey));
+  }
+
+  MockAppPtr mock_app_;
+  application_manager::AppExtensionPtr vi_app_extension_ptr_;
+  std::shared_ptr<sync_primitives::Lock> app_set_lock_ptr_;
+  vehicle_info_plugin::VehicleInfoPlugin vi_plugin_;
+  application_manager_test::MockRPCHandler mock_rpc_handler_;
+  resumption::LastStateWrapperPtr mock_last_state_;
+};
+
+TEST_F(SubscribeVehicleRequestTest, Run_AppNotRegistered_UNSUCCESS) {
+  CommandPtr command(CreateCommandVI<SubscribeVehicleDataRequest>());
+  EXPECT_CALL(app_mngr_, application(_))
```

Comment:
Please check application id

Reply:
@LuxoftAKutsan check added in https://github.com/smartdevicelink/sdl_core/pull/3391/commits/c32519a9280675d19886de786afe15f756f0de31